enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,176 @@
+
+### Non-Goals
+- The enhancement will not describe the OCM user experience.
+- The enhancement will not provide a generic way to pull information from OCM into the cluster.
+- The process will not apply to disconnected clusters.
+
+## Proposal
+
+The Insights operator already communicates with the OCM API (see [Insights Operator pulling and exposing SCA certs from the OCM API](../insights/pulling-sca-certs-from-ocm.md). This is the proposed flow for updating the `pull-secret`:
+
+1. OCM provides an API endpoint that the cluster can call to check the current global updated `pull-secret`.
+2. User starts the migration process in the OCM by providing the `pull-secret` associated with the account to which the ownership will be transferred to.
+   * The details of whether the user provides the pull secret as a means of verification, or whether a request/approval workflow is implemented, is left to the OCM/UI flow. What is pertinent here is that there is an endpoint available to a cluster entity to fetch the new `pull-secret`.
+3. OCM validates the `pull-secret` for the new account and makes it available to the cluster for some time period i.e. the new API will now return the new `pull-secret` provided by the user.
+4. The Insights operator calls this API periodically, fetches the updated `pull-secret` and compares it with the current global `pull-secret` on the cluster (only entries available in both pull-secrets are compared).
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * Insights operator only replaces the entries provided in the updated `pull-secret` from the OCM API.
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure.
+
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
```

Comment:
@mrunalp @markturansky should we add API details here once MR 2716 has merged?

Reply:
@aravindhp  yes, and I think that MR can be wrapped up today. I'll be doing a final review with @abhgupta this afternoon.

Reply:
Yes, sounds good :+1:

Reply:
I mentioned the `ClusterTransfer` API here and added the link to the MR.